vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/shared/internal/DisableOnClickController.java

```diff
@@ -52,13 +53,16 @@ public class DisableOnClickController<C extends Component & HasEnabled & ClickNo
      * @param component
      *            the component to control, not {@code null}
      */
+    @SuppressWarnings({ "unchecked", "rawtypes" })
     public DisableOnClickController(C component) {
         this.component = Objects.requireNonNull(component);
-        component.addClickListener(event -> {
-            if (isDisableOnClick()) {
-                component.setEnabled(false);
-            }
-        });
+
+        ComponentUtil.addListener(component, ClickEvent.class,
+                (ComponentEventListener) (event -> {
+                    if (isDisableOnClick()) {
+                        component.setEnabled(false);
+                    }
+                }));
```

Comment:
Neither `MenuItemBase` nor `GridMenuItem` has `ClickNotifier`. I figure it's not really required for the controller either, I used `ComponentUtil` for adding the listener instead.